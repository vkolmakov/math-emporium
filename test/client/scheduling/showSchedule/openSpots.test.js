import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import OpenSpots from '../../../../client/schedulingApp/showSchedule/components/openSpots';

describe('OpenSpots component', () => {
    const startDate = moment('2017-01-02 00:00:00');
    const now = moment('2017-01-03 06:00:00');
    it('renders an appropriate error message if location is missing', () => {
        let component = renderer.create(
            <OpenSpots isLocationSelected={false}
                       isCourseSelected={false}
                       startDate={startDate}
                       now={now}
                       openSpots={[]} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        component = renderer.create(
            <OpenSpots isLocationSelected={false}
                       isCourseSelected={true}
                       startDate={startDate}
                       now={now}
                       openSpots={[]} />
        );

        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders an appropriate error message if course is not selected but location is', () => {
        const component = renderer.create(
            <OpenSpots isLocationSelected={true}
                       isCourseSelected={false}
                       startDate={startDate}
                       now={now}
                       openSpots={[]} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders a spinner if both location and course are selected, but openSpots are empty', () => {
        const component = renderer.create(
            <OpenSpots isLocationSelected={true}
                       isCourseSelected={true}
                       startDate={startDate}
                       now={now}
                       openSpots={[]} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('correctly partitions and renders open spots', () => {
        const openSpots = [
            { count: 2, time: 540, weekday: 3 },
            { count: 2, time: 600, weekday: 2 },
            { count: 1, time: 600, weekday: 4 },
            { count: 5, time: 660, weekday: 4 },
        ];

        const component = renderer.create(
            <OpenSpots isLocationSelected={true}
                       isCourseSelected={true}
                       startDate={startDate}
                       now={now}
                       openSpots={openSpots} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('applies correct classes to empty and expired spots', () => {
        const openSpots = [
            { count: 2, time: 540, weekday: 1 },
            { count: 0, time: 600, weekday: 1 },
            { count: 0, time: 600, weekday: 2 },
        ];

        const component = renderer.create(
            <OpenSpots isLocationSelected={true}
                       isCourseSelected={true}
                       startDate={startDate}
                       now={now}
                       openSpots={openSpots} />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });
});
