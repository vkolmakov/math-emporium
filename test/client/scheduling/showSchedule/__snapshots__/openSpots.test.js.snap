exports[`OpenSpots component applies correct classes to empty and expired spots 1`] = `
<div
  className="open-spots-display">
  <div
    className="weekday">
    <p>
      Mon, 01/02
    </p>
    <div
      className="expired-spot">
      <span
        onClick={[Function]}>
        09:00 am: 2 available
      </span>
    </div>
    <div
      className="expired-spot">
      <span
        onClick={[Function]}>
        10:00 am: none available
      </span>
    </div>
  </div>
  <div
    className="weekday">
    <p>
      Tue, 01/03
    </p>
    <div
      className="closed-spot">
      <span
        onClick={[Function]}>
        10:00 am: none available
      </span>
    </div>
  </div>
</div>
`;

exports[`OpenSpots component correctly partitions and renders open spots 1`] = `
<div
  className="open-spots-display">
  <div
    className="weekday">
    <p>
      Tue, 01/03
    </p>
    <div
      className="open-spot">
      <span
        onClick={[Function]}>
        10:00 am: 2 available
      </span>
    </div>
  </div>
  <div
    className="weekday">
    <p>
      Wed, 01/04
    </p>
    <div
      className="open-spot">
      <span
        onClick={[Function]}>
        09:00 am: 2 available
      </span>
    </div>
  </div>
  <div
    className="weekday">
    <p>
      Thu, 01/05
    </p>
    <div
      className="open-spot">
      <span
        onClick={[Function]}>
        10:00 am: 1 available
      </span>
    </div>
    <div
      className="open-spot">
      <span
        onClick={[Function]}>
        11:00 am: 5 available
      </span>
    </div>
  </div>
</div>
`;

exports[`OpenSpots component renders a spinner if both location and course are selected, but openSpots is a Left of loading message 1`] = `
<div
  className="open-spots-message">
  <div
    className="loading" />
</div>
`;

exports[`OpenSpots component renders an appropriate error message if course is not selected but location is 1`] = `
<div
  className="open-spots-message">
  <h1>
    Select a Course
  </h1>
</div>
`;

exports[`OpenSpots component renders an appropriate error message if location is missing 1`] = `
<div
  className="open-spots-message">
  <h1>
    Select a Location
  </h1>
</div>
`;

exports[`OpenSpots component renders an appropriate error message if location is missing 2`] = `
<div
  className="open-spots-message">
  <h1>
    Select a Location
  </h1>
</div>
`;

exports[`OpenSpots component renders an error message if both location and course are selected, but openSpots is a Left of any other message 1`] = `
<div
  className="open-spots-message">
  <h1>
    No ducks here.
  </h1>
</div>
`;
